web/packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -179,35 +207,11 @@ const useDesktopPlayer = ({ clusterId, sid }) => {
     setStatusText(info);
   }, []);
 
-  const clientOnClientScreenSpec = useCallback(
```

Comment:
This is no longer needed thanks to the CSS property `object-fit: 'scale-down'`. Please note that the scaling behavior has changed a bit. Previously, if the browser window was larger than the recording, we scaled the recording up. I'm not sure if it's the right thing to do, it could result in a low quality, pixelated image in some scenarios. 
Instead, now the physical size of a recording won't exceed its resolution. I don't know if the previous behavior was a bug or a feature, but I can bring it back with a different value of `object-fit` :)

Also, a quick preview of a bug that we had before, caused by manual scaling:


https://github.com/user-attachments/assets/95ca1470-dd30-40bb-923e-234203544f1a

Reply:
I'll have to double check, but I think there is also a usecase where we don't want the canvas to change resolution at all in order to preserve some special aspect ratio on the remote device. however, since that wasn't a thing on the client side before (i think) then its still probably handled on the server side. 

i could also be misremembering anyway, so I'll look for it

Reply:
I believe the new implementation should be no worse than the previous one at handling it, but maybe I'm missing something. If you will have more details about it please let me know!